tasks.py

```diff
@@ -596,3 +611,78 @@ def _view_messages(queue_type, limit):
             if 'MessageAttributes' in message and 'FailedReason' in message['MessageAttributes']:
                 print("{}: {}".format('ReportedError', message['MessageAttributes']['FailedReason']['StringValue']))
             print()
+
+
+def _schedule_csv_to_json(csv_name, json_name, year=None, semester=None):
+    if year is None:
+        year = str(datetime.now().year)
+    if semester is None:
+        month = datetime.now().month
+        if month < 6:
+            semester = "02"
+        elif month < 9:
+            semester = "03"
+        else:
+            semester = "01"
+    elif semester not in ["01", "02", "03"]:
+        print("Semester must be '01' for fall. '02' for winter, or '03' for summer.")
+        return
+
+    csv_file = open(csv_name, "r")
+    json_file = open(json_name, "w")
+
+    fieldnames = tuple(csv_file.readline().split(',')[:-1])
+
+    reader = csv.DictReader(csv_file, fieldnames)
+
+    data = {}
+
+    for row in reader:
+        del row[None]
+
+        # filter out empty fields for dynamo
+        course = {}
+        for key, val in row.items():
+            if val != '':
+                course[key] = val
+
+        if 'https://zoom.us' not in row['Links']:
+            # not a zoom course (probably Adobe)
+            continue
+        else:
+            zoom_series_id = row['Links'].split('/')[-1]
+
+        opencast_series_id = year + semester + row["CRN"].strip()
+        opencast_subject = row["Subject"].strip()\
+                           + (" S-" if semester == "03" else " E-") \
+                           + row["Course Number"].strip()
+
+        course['opencast_series_id'] = opencast_series_id
+        course['opencast_subject'] = opencast_subject
+        course['zoom_series_id'] = zoom_series_id
+        course['Days'] = [x.strip() for x in course['Day'].split("/")]
+        del course['Day']
+
+        data[zoom_series_id] = course
+
+    json.dump(data, json_file)
```

Comment:
you never know when a human might want to read this. i always use `indent=2` unless i'm in some kind of legit bandwidth or storage constraint.

Reply:
Fixed.